### ISSUE TITLE
Why microphone access permission?

### ISSUE THREAD

**Issue by java-py-c-cpp-js** [dtsudo/Chess-Comp-Stomp-with-Hacks-Electron-Version#2]
The current [Flatpak package](https://flathub.org/apps/io.github.dtsudo.ChessCompStompWithHacks) is setup with
> Microphone access
> Can listen using microphones without asking permission

![screenshot](https://github.com/user-attachments/assets/abf7d11f-1e63-4280-949d-66af8d46b8c9)

**Comment by dtsudo** [comment]
Thanks for filing the issue!

For what it's worth, although the game has microphone access, it doesn't actually listen to the microphone or do anything with this access.

As far as why this permission is present, flatpak applications have a set of permissions that they can request. Per [the documentation](https://docs.flatpak.org/en/latest/sandbox-permissions.html#standard-permissions), the most relevant one is `--socket=pulseaudio`:

> `--socket=pulseaudio` - Access to PulseAudio, includes sound input (mic), sound output/playback, MIDI and ALSA sound devices in `/dev/snd`

Most games need this permission, because this is what allows the game to play sound/music. The microphone access comes bundled with this permission though.

Thus, we see that basically any flatpak application that needs to play sound will have microphone access, even though the vast majority of such applications don't actually need the mic -- e.g.:

* [SuperTux](https://flathub.org/apps/org.supertuxproject.SuperTux)
* KDE's chess program [KNights](https://flathub.org/apps/org.kde.knights)
* Even simple programs like KDE's solitaire game has mic access - https://flathub.org/apps/org.kde.kpat

In short, games need access to PulseAudio to play sound, and that permission also includes access to microphones. Some people have suggested more granular permission levels -- e.g. https://github.com/flatpak/flatpak/issues/3425 -- but no such feature has been released yet for flatpaks.

It's worth noting that the game doesn't have Internet access, so the sandbox is still pretty solid. There's no way to exfiltrate data without network access, so certainly no data (whether mic data or otherwise) will leave your device.
